src/runtime/Module.cpp

```diff
@@ -227,6 +233,11 @@ Instance* Module::instantiate(ExecutionState& state, const ExternVector& imports
         tagIndex++;
     }
 
+    // All memories are resolved, enque them.
+    for (size_t i = 0; i < numberOfMemoryTypes(); i++) {
+        targetBuffers[i].enque(instance->m_memories[i]);
+    }
```

Comment:
As here, each Memory has one `TargetBuffer` and initialized all at once when the Instance is created.
And all `TargetBuffer` is dequed when the Instance is destoyed.
So, I'm not sure why `TargetBuffer` has `next` reference in itself.

Reply:
When a Memory grows, it needs to update all TargetBuffers for all Instances. It has a chain list of TargetBuffers, and goes through them. Normally the Instance knows its Memories, but Memories don't know which Instance uses them.

Reply:
I see. I missed the point that several `TargetBuffers` could be created for one `Memory` when the Memory is shared in multiple Instances.